hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/HashJoinCollectP.java

```diff
@@ -44,11 +45,11 @@ protected boolean tryProcess0(@Nonnull Object item) {
         E e = (E) item;
         K key = keyFn.apply(e);
         V value = projectFn.apply(e);
-        V previous = map.put(key, value);
-        if (previous != null) {
-            throw new IllegalStateException(String.format("Duplicate values for key '%s': '%s' and '%s'",
-                    key, previous, value));
-        }
+
+        map.merge(key, new ArrayList<>(singletonList(value)), (prev, curr) -> {
```

Comment:
This should be:
```java
map.computeIfAbsent(key, k -> new ArrayList<>())
   .add(value)
```

Reply:
It can be, but it also can be like in the commit, see https://www.nurkiewicz.com/2019/03/mapmerge-one-method-to-rule-them-all.html

Reply:
You unconditionally create garbage with `map.merge()`. @viliam-durina `k -> new ArrayList<>()` should also be a constant, right?

Reply:
It allocates a new ArrayList regardless if it's needed or not.

Reply:
@mtopolnik yes
@cangencer the current version does, the updated doesn't.
@TomaszGaweda it was correct as it was, it only created more GC litter and also was harder to read

Reply:
@viliam-durina got it, thanks :)